Remove periodic-ci workflow

gurevichdmitry opened the pull request:
### Summary of your changes

This PR removes the unused workflow `periodic-ci.`

### Screenshot/Data
<!--
If this PR adds a new feature, please add an example screenshot or data (findings json for example).
-->


### Related Issues
<!--
- Related: https://github.com/elastic/security-team/issues/
- Fixes: https://github.com/elastic/security-team/issues/
-->

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?

- [ ] Generate rule metadata using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rules-metadata)
- [ ] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/cloudbeat/tree/main/security-policies/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

orouz commented:
@gurevichdmitry i think we can remove the `Build` command on the magefile, as it was only used in this workflow. maybe even some more magefile cleanup would be possible because of it.

gurevichdmitry commented:
> @gurevichdmitry i think we can remove the `Build` command on the magefile, as it was only used in this workflow. maybe even some more magefile cleanup would be possible because of it.

@orouz, I removed the `Build` command and the `build.sh` script, which had no references. Additionally, I updated the documentation where mage build was referenced.
